### PR TITLE
Update settings to latest version of gitlab api.

### PR DESCRIPTION
**Pull request by jabby**
According to the documentation there is many new settings available.
This commit update all settings and documentation to the latest version.

It's actually a work in progress as there is few TODOs in the code.

Basically the first verification to do is to validate that this modification is retro-compatible
with olders version of library.

I will update this PR next day. Feel free to comment my work in progress.

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/models/Setting.java [hunk 1]
```diff
@@ -67,54 +108,60 @@
...
-    CLIENTSIDE_SENTRY_DSN(String.class),
-
+    COMMIT_EMAIL_HOSTNAME(String.class),
+    
     /**
-     * (If enabled, requires: clientside_sentry_dsn) Enable Sentry error reporting
-     * for the client side.
+     * Container Registry token duration in minutes.
      */
-    CLIENTSIDE_SENTRY_ENABLED(Boolean.class),
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/Setting.java]: This should be deprecated not removed.

File: src/main/java/org/gitlab4j/api/models/Setting.java [hunk 2]
```diff
@@ -67,54 +108,60 @@
...
+     * available to projects if the project namespace’s plan includes the feature
+     * or if the project is public.
      */
     CHECK_NAMESPACE_PLAN(Boolean.class),
 
     /**
-     * required by: clientside_sentry_enabled Clientside Sentry Data Source Name.
+     * Custom hostname (for private commit emails).
      */
-    CLIENTSIDE_SENTRY_DSN(String.class),
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/Setting.java]: This should be deprecated not removed.

File: src/main/java/org/gitlab4j/api/models/Setting.java [hunk 3]
```diff
@@ -217,485 +307,629 @@
...
     RESTRICTED_VISIBILITY_LEVELS(String[].class),
 
     /**
-     * The minimum allowed bit length of an uploaded RSA key. Default is 0 (no
-     * restriction). -1 disables RSA keys.
+     * The minimum allowed bit length of an uploaded RSA key. Default is 0 (no restriction).
+     * -1 disables RSA keys.
      */
-    RSA_KEY_RESTRICTION(Integer.class),
+    RSA_KEY_RESTRICTION(Boolean.class),
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/Setting.java]: The documentation indicates that ```0``` or ```-1``` is returned, why change this to ```Boolean.class````

File: src/main/java/org/gitlab4j/api/models/Setting.java (hunk 2 above)
Comment (jabby) [reply src/main/java/org/gitlab4j/api/models/Setting.java]: I updated my patch. It was a mistake.

File: src/main/java/org/gitlab4j/api/models/Setting.java (hunk 1 above)
Comment (jabby) [reply src/main/java/org/gitlab4j/api/models/Setting.java]: I updated my patch. It was a mistake.

File: src/main/java/org/gitlab4j/api/models/Setting.java (hunk 3 above)
Comment (jabby) [reply src/main/java/org/gitlab4j/api/models/Setting.java]: Good catch. My bad.

**Comment by jabby** [comment]
@gmessner I updated the change accordingly to your comments. I also updated few more things and dig a little for fields with no documentation. I updated javadoc when I found some useful informations. 

There is one task to do before merging this patch. See https://github.com/gitlab4j/gitlab4j-api/pull/462/files#diff-999bbdddeac9c1f965364d46f7c74cdeR88
I don't know which type we should use here.

**Comment by gmessner** [comment]
@jabby 
I have a way to have it support both ```String``` and ```String[]```, but it would be more work for me to explain how to do it rather than do it myself.

My suggestion would be for you to finish everything else and let me know it is ready to be merged.  Once merged I will add support for the dual type and keep you updated on how it was accomplished.

**Comment by jabby** [comment]
For my point of view, except about the support of `String` and `String[]`, this patch can be merged.
So feel free to merged and provide the additional change. I will check your modification and learn how you manage this case.

**Comment by gmessner** [comment]
@jabby 
I am currently traveling, will be back late 11/6/2019 will merge then and add support for the ```String/String[]``` dual type.

**Comment by jabby** [comment]
Ok no problem. Have a nice travel.

**Pull request merged**